users/postgres/database.go

```diff
@@ -0,0 +1,60 @@
+package postgres
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/opentracing/opentracing-go"
+)
+
+var _ DatabaseMiddleware = (*databaseMiddleware)(nil)
+
+type databaseMiddleware struct {
+	db *sqlx.DB
+}
+
+// DatabaseMiddleware provides a database interface
+type DatabaseMiddleware interface {
```

Comment:
Do the same thing for this middleware.

Reply:
These middlewares are currently the same.  I'm wondering if they shouldn't be put in a common location, maybe in a folder next to the `Logger` middleware.  I could see the tag values changing depending on where the interface is used so there would need to be a way to override the `addSpanTags` function.

Reply:
@nwneisen we have an issue for creating the packages for the shared code. I recommend that we leave it for another PR.

Reply:
Probably related to https://github.com/mainflux/mainflux/issues/666 which is a placeholder for a general package reorganization.